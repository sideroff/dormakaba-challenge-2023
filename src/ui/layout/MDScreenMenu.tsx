import List from '@mui/material/List';
import { NavbarListItemButton } from '@/ui/layout/NavbarListItemButton';

interface MDScreenMenuProps {
  onNavMenuClose: () => void;
}

export function MDScreenMenu({ onNavMenuClose }: MDScreenMenuProps) {
  return (
    <List
      onClick={onNavMenuClose}
      sx={{
        display: { xs: 'none', md: 'flex' },
        flexDirection: 'row',
        flexGrow: 1,
      }}
    >
      <NavbarListItemButton label="Doors" href="/doors" />
    </List>
  );
}
