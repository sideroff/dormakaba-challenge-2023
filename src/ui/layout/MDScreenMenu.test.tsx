import { render } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import { MDScreenMenu } from './MDScreenMenu';

jest.mock('next/router', () => require('next-router-mock'));

describe('MDScreenMenu', () => {
  it('should render correctly', () => {
    mockRouter.push('/test');

    const handleCloseNavMenu = jest.fn();

    const { container } = render(
      <MDScreenMenu onNavMenuClose={handleCloseNavMenu} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
