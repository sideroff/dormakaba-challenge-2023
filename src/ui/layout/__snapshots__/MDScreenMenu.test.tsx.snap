// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDScreenMenu should render correctly 1`] = `
<ul
  class="MuiList-root MuiList-padding css-hmqy6w-MuiList-root"
>
  <a
    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-72rlw3-MuiButtonBase-root-MuiListItemButton-root"
    href="/doors"
    tabindex="0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Doors
    </p>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </a>
</ul>
`;
